### PR TITLE
Enable inspector tests for ChakraCore

### PR DESCRIPTION
**Pull request by kfarnung**
##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines)

##### Affected core subsystem(s)

build, test

### CODE REVIEW

**Comment by kfarnung** [comment]
~~CI: https://ci.nodejs.org/job/chakracore-test-pull-request/38/~~
New CI: https://ci.nodejs.org/job/chakracore-test-pull-request/42/

File: test/inspector/inspector.status [hunk 1]
```diff
@@ -9,4 +9,3 @@ prefix inspector
 [$system==win32]
 
 [$jsEngine==chakracore]
```
Comment (digitalinfinity) [inline test/inspector/inspector.status]: Why not remove this section altogether and have it match upstream?

File: test/parallel/test-whatwg-url-setters.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ const { test, assert_equals } = common.WPT;
 const additionalTestCases = require(
     path.join(common.fixturesDir, 'url-setter-tests-additional.js'));
 
-if (!common.hasIntl) {
+if (!common.hasIntl || common.isChakraEngine) {
```
Comment (digitalinfinity) [inline test/parallel/test-whatwg-url-setters.js]: Isn't this too broad? Like Intl should be available on Windows, no?

**Review by digitalinfinity — APPROVED** [review]

File: test/parallel/test-whatwg-url-setters.js (hunk 2 above)
Comment (kfarnung) [reply test/parallel/test-whatwg-url-setters.js]: I can take another look, but this was an issue on Windows in my testing.

File: test/inspector/inspector.status (hunk 1 above)
Comment (kfarnung) [reply test/inspector/inspector.status]: Easy enough, I'll do that.

**Comment by kunalspathak** [comment]
LGTM. Agree with @digitalinfinity feedback though.

File: test/parallel/test-whatwg-url-setters.js (hunk 2 above)
Comment (kfarnung) [reply test/parallel/test-whatwg-url-setters.js]: Odd, here's the failure on see on Windows:

```
=== release test-whatwg-url-setters ===
Path: parallel/test-whatwg-url-setters
TypeError: Object doesn't support property or method 'ToString'
   at href.set (internal/url.js:307:7)
   at Anonymous function (E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js:121:27)
   at _tryBlock (assert.js:503:5)
   at _throws (assert.js:522:3)
   at throws (assert.js:552:3)
   at Anonymous function (E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js:121:7)
   at Module.prototype._compile (module.js:582:5)
   at Module._extensions[.js] (module.js:593:3)
   at Module.prototype.load (module.js:516:3)
   at tryModuleLoad (module.js:466:5)
Command: E:\GitHub\node-chakracore\Release\node.exe E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js
```

I'll investigate further.

**Comment by kfarnung** [comment]
This is the first part of #227.

**Comment by kfarnung** [comment]
This should also resolve #226

File: src/inspector_io.cc [hunk 3]
```diff
@@ -1,476 +1,445 @@
-#include "inspector_io.h"
```
Comment (kfarnung) [inline src/inspector_io.cc:1]: This is the result of `git checkout nodejs/master src/inspector_io.cc`.  At some point the line endings must have gotten messed up.  The diff here just removes the UTF-8/UTF-16 conversion workarounds that I added for inspector.

**Pull request merged**